staging/src/k8s.io/client-go/tools/watch/retrywatcher.go

```diff
@@ -126,6 +126,22 @@ func (rw *RetryWatcher) doReceive() (bool, time.Duration) {
 			return false, 0
 		}
 
+		// Check if the watch failed due to the client not having permission to watch the resource or the credentials
+		// being invalid (e.g. expired token).
+		if apierrors.IsForbidden(err) || apierrors.IsUnauthorized(err) {
+			// Add more detail since the forbidden message returned by the Kubernetes API is just "unknown".
+			klog.ErrorS(err, msg+": ensure the client has watch permissions on the resource")
+
+			statusErr := err.(apierrors.APIStatus).Status()
```

Comment:
thinking more about it
actually, to be really on the safe side and avoid regression in the past, let's switch to
```
if apistatus, ok := err.(apierrors.APIStatus); ok {
  statusErr := apistatus.Status()
  _ = rw.send(...)
} else 
  // this should never happen as apierrors are handling only apierrors.APIStatus
  klog.Errors( ....)
```